lib/spack/spack/package.py

```diff
@@ -714,14 +714,14 @@ def env_path(self):
         if self.stage.source_path is None:
             return None
         else:
-            return os.path.join(self.stage.source_path, 'spack-build.env')
+            return os.path.join(self.stage.path, 'spack-build.env')
 
     @property
     def log_path(self):
         if self.stage.source_path is None:
```

Comment:
This function should probably be eliminated. If you disagree, then this test should likely be `if self.stage.path is None` or `if not self.stage.path`.

Reply:
I'm not quite sure why you think `env_path()` should be eliminated? I can certainly change the test.

Reply:
This is the `log_path` function. Between `log_path` and `build_log_path` there should likely only be one. This function appears to be inconsistent but `build_log_path` does not, so I suggest removing this one. See also: https://github.com/spack/spack/pull/8431#discussion_r224239598

Reply:
Ah, OK: the patch lines were part of `env_path()` according to the diff label, so that's what I thought you were referring to.

Reply:
If it seems difficult to remove on further inspection, I'll take your word for it. That being said `grep` doesn't find any references in the main repo (`var/spack/repos/builtin/`) or for any of the build system subclasses of `Package`, and the logic looks similar.

IMO it's not worth chasing too much, it just looked like an opportunity to simplify things.

However, if it's going to stick around, the test should be `if self.stage.path is None` or `if not self.stage.path`.